### PR TITLE
bug fix: fix agent_logging test

### PR DESCRIPTION
**Pull request by cheng-tan**
## Why are these changes needed?
agent_logging test checks the response from the LLM, which might fail in the CI for some configs due to too many requests. This fix checks the error is logged for failed request or the correct chat completion response is logged for successful request.

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

File: test/agentchat/test_agent_logging.py [hunk 1]
```diff
@@ -110,7 +110,11 @@ def test_two_agents_logging(db_connection):
         assert first_request_role == "system"
 
         response = json.loads(row["response"])
-        assert "choices" in response and len(response["choices"]) > 0
+
+        if "response" in response:  # config failed or response was empty
```
Comment (ekzhu) [inline test/agentchat/test_agent_logging.py:114]: In a sense the purpose is to validate the response is a valid chat completion response object. Do we know that "error_code" and "choices" check is sufficient? i.e., check this JSON is a valid `openai.ChatCompletion` object. Or do we not need to worry about the validation?

**Review by ekzhu — APPROVED** [review]

**Comment by ekzhu** [comment]
Can you move this PR to a branch from upstream branch so the open ai tests can run?

File: test/agentchat/test_agent_logging.py [hunk 2]
```diff
@@ -110,7 +110,11 @@ def test_two_agents_logging(db_connection):
...
+            assert "error_code" in response["response"] or response["response"] is None
```
Comment (sonichi) [inline test/agentchat/test_agent_logging.py:115]:
```suggestion
            assert response["response"] is None or "error_code" in response["response"]
```

**Comment by sonichi** [comment]
To run the openai test, please make the PR from the upstream repo.

**Pull request merged**